pandas/core/groupby/generic.py

```diff
@@ -1826,4 +1826,46 @@ def nunique(self, dropna: bool = True) -> DataFrame:
             self._insert_inaxis_grouper_inplace(results)
         return results
 
+    @Appender(DataFrame.idxmax.__doc__)
+    def idxmax(self, axis=0, skipna: bool = True):
+        axis = DataFrame._get_axis_number(axis)
+        numeric_only = None if axis == 0 else False
+
+        def func(df):
+            # NB: here we use numeric_only=None, in DataFrame it is False GH#38217
+            res = df._reduce(
+                nanops.nanargmax,
+                "argmax",
+                axis=axis,
+                skipna=skipna,
+                numeric_only=numeric_only,
+            )
+            indices = res._values
+            index = df._get_axis(axis)
+            result = [index[i] if i >= 0 else np.nan for i in indices]
```

Comment:
isn't this a .take()?

Reply:
it's take-like, but using take directly raises with Int64Index.  This is almost identical to what we have in DataFrame.idxmax/idxmin.